src/frontend/interfaces/page-json.interface.ts

```diff
@@ -16,4 +19,9 @@ export interface PageJSON {
    * Page icon
    */
   icon?: string;
+
+  /**
+   * Page visibility
+   */
+  isVisible: boolean | IsFunction;
```

Comment:
`PageJSON` is something returned to the frontend from the backend so it cannot be a function (`IsFunction`) and has to be pre-evaluated.

`pagesToStore` would have to do something like:
```
  Object.entries(pages).map(([key, adminPage]) => {
      let isVisible = true;
      if (adminPage.isVisible) {
         if (typeof adminPage.isVisible === 'function') {
           isVisible = adminPage.isVisible(currentAdmin)
         } else {
           isVisible = adminPage.isVisible
         }
      }
      return {
        name: key,
        component: adminPage.component,
        icon: adminPage.icon,
        isVisible,
      }
  })
```